### ISSUE TITLE
official pr-downloader build doesn't contain https support -> cannot download

### ISSUE THREAD

**Issue by gajop** [spring/pr-downloader#112]
Chobby users on Windows have reported download issues with maps in latest engine, and @azaremoth was kind enough to test stand-alone pr-downloader, which indeed has the same problem.

Compare one of the newest prds: 

> 104.0.1-972-gb2624e5 maintenance
```
engine\104.0.1-972-gb2624e5 maintenance>pr-downloader.exe --download-map "Red Comet"
pr-downloader 0.7-496-g813ac02
[Info] ../../tools/pr-downloader/src/FileSystem/FileSystem.cpp:194:setWritePath():Using filesystem-writepath: C:\Users\jan_b\OneDrive\Dokumente\My Games\Spring
[Info] ../../tools/pr-downloader/src/pr-downloader.cpp:173:DownloadSetConfig():Free disk space: 4648 MB
[Info] ../../tools/pr-downloader/src/Downloader/CurlWrapper.cpp:51:DumpVersion():libcurl 7.24.0 (null)
[Info] ../../tools/pr-downloader/src/Downloader/CurlWrapper.cpp:96:ValidateCaFile():Downloading https://springrts.com/dl/cacert.pem
[Error] ../../tools/pr-downloader/src/Downloader/Http/HttpDownloader.cpp:460:processMessages():CURL error(1:1): Unsupported protocol 0 (https://springrts.com/dl/cacert.pem)

[Error] ../../tools/pr-downloader/src/Downloader/Http/HttpDownloader.cpp:87:DownloadUrl():Error in curl Unsupported protocol (Protocol https not supported or disabled in libcurl)
[Error] ../../tools/pr-downloader/src/Downloader/Http/HttpDownloader.cpp:207:search():Error downloading https://api.springfiles.com/json.php?category=map&torrent=true&springname=Red Comet
[Error] ../../tools/pr-downloader/src/main.cpp:178:main():No map found for Red Comet
[Info] ../../tools/pr-downloader/src/main.cpp:223:main():Download complete!
```

With an older one:
> 104.0.1-287-gf7b0fcc maintenance
```
Spring_DEV>pr-downloader.exe --download-map "Red Comet"
pr-downloader 0.7-394-gb3e0e6a
[Info] Using filesystem-writepath: C:\Users\jan_b\OneDrive\Dokumente\My Games\Spring
[Info] Free disk space: 4628 MB
[Info] Server refused ranges
[Info] piece finished==========================] 2656510/2656510

[Info] Download complete!
```

**Comment by abma** [comment]
https://github.com/spring/spring-lxc/blob/master/scripts/cfg/install/install-mxe.sh

yes. only gcc  & boost are installed via mxe.

**Comment by abma** [comment]
i'm not 100% sure... it was a long time ago :)

**Comment by abma** [comment]
please leave this issue open: pr-downloader should force https support and show an error when disabled. afaik this is not the case.

**Comment by ashdnazg** [comment]
x32 according to the number of digits in the git version

**Comment by gajop** [comment]
Ideally fix whichever is linked into spring and distributed as standalone executable. Probably x86 still...

**Comment by ashdnazg** [comment]
ok, libcurl wasn't changed for 7 years, I don't think it's me
https://github.com/spring/mingwlibs/tree/master/lib

**Comment by abma** [comment]
i wonder why it works with the old version :thinking:

**Comment by abma** [comment]
ugh, trivial: the api link is hard coded :-|

**Comment by silentwings** [comment]
I think this issue has become the main block to requiring email verification (because it prevents projects using the unmaintained SWL from switching to chobby).

**Comment by abma** [comment]
partly fixed via https://github.com/spring/spring-lxc/commit/dfc8dd1bc7802d3aadf2dbcc18e53c4836ddd5b1

some other stuff broke, too, i.e. java ai libs + linking to curl (missing linking flags) when using mxe libs to compile

**Comment by abma** [comment]
at least the provided libcurl.dll should support https: can you please test if the current build of pr-downloader works?


https://springrts.com/dl/buildbot/default/develop/104.0.1-1493-g53e5977/win32/

note: it requires libcurl.dll to work.

**Comment by azaremoth** [comment]
Rapid Downloads are working. Map dowloads are not.

D:\Spring_development>pr-downloader.exe --download-map "Red Comet"
pr-downloader 0.7-503-gb091bee
[Info] ../../tools/pr-downloader/src/FileSystem/FileSystem.cpp:194:setWritePath():Using filesystem-writepath: C:\Users\jan_b\OneDrive\Dokumente\My Games\Spring
[Info] ../../tools/pr-downloader/src/pr-downloader.cpp:173:DownloadSetConfig():Free disk space: 14652 MB
[Info] ../../tools/pr-downloader/src/Downloader/CurlWrapper.cpp:51:DumpVersion():libcurl 7.64.0 Schannel
[Info] ../../tools/pr-downloader/src/Downloader/CurlWrapper.cpp:40:DumpTLSInfo():SSL backend #0: 'schannel' (ID: 8)
[Info] ../../tools/pr-downloader/src/Downloader/CurlWrapper.cpp:96:ValidateCaFile():Downloading https://springrts.com/dl/cacert.pem

[Error] ../../tools/pr-downloader/src/Downloader/Http/HttpDownloader.cpp:87:DownloadUrl():Error in curl SSL peer certificate or SSH remote key was not OK (schannel: CertGetCertificateChain trust error CERT_TRUST_IS_UNTRUSTED_ROOT)
[Error] ../../tools/pr-downloader/src/Downloader/Http/HttpDownloader.cpp:207:search():Error downloading https://api.springfiles.com/json.php?category=map&torrent=true&springname=Red Comet
[Error] ../../tools/pr-downloader/src/main.cpp:178:main():No map found for Red Comet
[Info] ../../tools/pr-downloader/src/main.cpp:223:main():Download complete!

**Comment by abma** [comment]
@azaremoth:

does "iexplore https://springrts.com/dl/" work without certificate warning/error?

**Comment by azaremoth** [comment]
yes - the access works

**Comment by abma** [comment]
next try, does this version work for you?
https://springrts.com/dl/buildbot/default/develop/104.0.1-1519-g4b3186a/win32/

(for me it worked)

**Comment by abma** [comment]
https://springrts.com/mantis/view.php?id=6145#c19859

**Comment by ForbodingAngel** [comment]
So not sure if this is helpful
http://logs.springrts.com/logfiles/65/
I updated evo and it's working pretty well on 1208. I have the alpha test host using that engine version and this is what happened when I ran the launcher and joined it. So, it might be a version thing, not sure.

Specifically the map and engine and game all failed to dl
lines
41
51
60
264

fwiw I did try removing cacert just in case, but no dice
